src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -326,5 +335,21 @@ def getRequestSummaryWithJobInfo(self, requestName):
         self._updateRequestInfoWithJobInfo(requestInfo)
         return requestInfo
         
+    def getArchivedRequests(self):
+        """
+        get list of archived workflow in wmstats db.
+        """
+        
+        options = {"group_level": 1, "reduce": True}
+        
+        results = self.couchDB.loadView(self.couchapp, "allWorkflows", options = options)['rows']
+        requestNames = [x['key'] for x in results]
+        
+        workflowDict = self.reqDB.getStatusAndTypeByRequest(requestNames)
```

Comment:
Aren't we going to fetch way too many requests here? Maybe we could gather all the workflows newer than 3 months (is there a view by request creation date?), filter the archived ones and proceed with that small set only?